mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -98,6 +99,11 @@ def __init__(self, artifact_uri):
         self.run_relative_artifact_repo_root_path = (
             "" if run_artifact_root_path == artifact_repo_root_path else run_relative_root_path
         )
+        # Limit the number of threads used for artifact uploads, using at most `8` threads or
+        # 2 * the number of CPU cores available on the system (whichever is smaller)
+        num_cpus = os.cpu_count() or 4
+        num_artifact_workers = min(num_cpus * 2, 8)
+        self.thread_pool = ThreadPoolExecutor(max_workers=num_artifact_workers)
```

Comment:
Without being very knowledgeable thread pools and their resource overhead, I feel like 8 is a big improvement purely because we currently have no parallelization. Not sure if there's a good reason to overdo it from the get-go

Reply:
I took a look at the average number of artifact files on Databricks, and the p25 - p75 number of files is 3, while the p95 is 4. The p99 is over 2000. `min(8, 2 * cpus)` seems okay given that knowledge.